### ISSUE TITLE
Clarify license

### ISSUE THREAD

**Issue by arnout** [mono/libgdiplus#375]
The license of the libgdiplus project is not entirely clear:

- All source files have a MIT/X11 license header.
- The Mono project FAQ says "The Mono runtime, compilers and tools and most of the class libraries are licensed under the MIT license."
- However, the LICENSE file says that it is LGPL-2.1 or MPL-1.1.

Here is my detailed analysis of the situation:

 Since 2004 already, there has been a LICENSE file with the text "Libgdiplus is licensed under the terms of the GNU Library GPL or the Mozilla Public License 1.1." and with the details of the MPL (but not the LGPL!) in MPL-1.1.html. However, no license headers in source files were updated.

 In 2009, a spec file was added that says
License:        LGPL v2.1 only; MOZILLA PUBLIC LICENSE (MPL/NPL); X11/MIT
This file was added by someone from Novell, and Novell is mentioned in all the copyright headers in the source files.

 I haven't traced exactly when, but at some point contributions started requiring copyright assignment to Xamarin. For contributions between the addition of the LICENSE file and the copyright assignments, I think the status is a little unclear.

 Since 2016, instead of a copyright assignment, there is a Contributor License Agreement which basically licenses the .NET Foundation to relicense the code under whatever license they want.

 In September 2017, the MPL-1.1 and the LGPL-2.1 license texts were added to the LICENSE file, and MPL-1.1.html was removed.

 And finally, libgdiplus is part of the Mono project, and the Mono licensing FAQ says: "The Mono runtime, compilers and tools and most of the class libraries are licensed under the MIT license." (but libgdiplus is not really part of the mono runtime, compilers, tools or class libraries; it's an independent library that is used by the Mono project, probably by nobody else).

 Bottom line:

- The license situation really hasn't changed since 2004.
- Based on the information in the source tarball, it should be (LGPL-2.1 OR MPL-1.1) AND MIT.
- But the intention is probably MIT.

To remove confusion, it would be best to update either the LICENSE file or the copyright headers in the source files.

**Comment by migueldeicaza** [comment]
It is MIT, we relicensed it all under MIT when Microsoft acquired Xamarin.   We forgot a few spots, like this one.
